app/lib/Server.scala

```diff
@@ -0,0 +1,252 @@
+      jsonTicket  <- elem.get("ticket")
+
+      row         <- jsonRow.asOpt[Int]
+      column      <- jsonSeat.asOpt[Int]
+      ticketType  <- jsonTicket.asOpt[String].flatMap(parseTicketType(_))
+
+      seat = Seat(row, column)
+
+    } yield (seat -> ticketType)
+
+    case _ => None
+  }
+
+  def nameValid(name: String): Boolean = name.forall(_.isLetter) && name.length >= 3
+
+  def surnameValid(surname: String): Boolean = {
+    val segments = surname.split("-")
+    segments.length <= 2 && segments.forall(w => w.forall(_.isLetter) && w.length >= 1)
+  }
+
+
+  // Processing of data from the database ---------------------------------------------------------
+  def getAvailableSeats(
+    screeningId:  ScreeningId,
+    roomId:       RoomId,
+  ): EitherT[Future, Error, AvailableSeats] = for {
+
+    takenSeats <- getTakenSeats(
+      screeningId,
+      roomId,
+    )
+    availableSeats = getAvailableSeatsArray(takenSeats.dim, takenSeats.seats)
+
+  } yield AvailableSeats(takenSeats.dim, availableSeats)
+
+  def getTakenSeats(
+    screeningId:  ScreeningId,
+    roomId:       RoomId,
+  ): EitherT[Future, Error, TakenSeats] = for {
+
+    dim           <- dbInterface.getRoomDimension(roomId)
+    reservations  <- dbInterface.getReservations(screeningId)
+    reservedSeats = reservations.flatMap(_.seats.keys)
+
+  } yield TakenSeats(dim, reservedSeats)
+
+
+  // Request serving ------------------------------------------------------------------------------
+  def serveListScreeningsRequest(
+    date: String,
+    from: String,
+    to:   String,
+  ): EitherT[Future, Error, JsValue] = {
+
+    val dateFrom  = s"$date $from"
+    val dateTo    = s"$date $to"
+
+    val tryFrom = Try(DateTimeFormat.forPattern("dd-MM-yyyy HH:mm").parseDateTime(dateFrom))
+    val tryTo   = Try(DateTimeFormat.forPattern("dd-MM-yyyy HH:mm").parseDateTime(dateTo))
+
+
+    for {
+      from  <- EitherT.fromOption[Future](tryFrom .toOption, InvalidParameters)
+      to    <- EitherT.fromOption[Future](tryTo   .toOption, InvalidParameters)
+
+      screenings <- dbInterface.getScreenings(from, to)
+      screeningsSorted = screenings.sortBy(info => (info.title, info.duration))
+      screeningInfos = screeningsSorted.map(screeningInfoBasic(_))
+    } yield Json.toJson(screeningInfos)
+  }
+
+  def serveScreeningInfoRequst(idStr: String): EitherT[Future, Error, JsObject] = for {
+
+    screeningId <- EitherT.fromEither[Future](stringToObjectID(idStr, InvalidScreeningId))
+    info        <- dbInterface.getScreeningInfo(screeningId)
+    moreInfo    <- screeningInfo(info)
+
+  } yield moreInfo
+
+  def validateReservation(
+    reservation: Reservation,
+  ): EitherT[Future, Error, DateTime] = for {
+
+    info        <- dbInterface.getScreeningInfo(reservation.screening)
+    takenSeats  <- getTakenSeats(reservation.screening, info.room)
+
+    requestedSeats = reservation.seats.keys.toList
+
+    // Not to late to reserve seats
```

Comment:
Maybe we could extract properly named methods instead of commenting each step?

Reply:
Yes.